controllers/routingctrl/reconcile_resources.go

```diff
@@ -97,33 +93,46 @@ func (r *Controller) exportService(ctx context.Context, target *unstructured.Uns
 		}
 	}
 
-	return r.propagateHostsToWatchedCR(target, publicHosts, externalHosts)
+	return r.propagateHostsToWatchedCR(ctx, target, publicHosts, externalHosts)
 }
 
-func (r *Controller) propagateHostsToWatchedCR(target *unstructured.Unstructured, publicHosts, externalHosts []string) error {
-	// Remove all existing routing addresses
-	metaOptions := []metadata.Option{
-		annotations.Remove(annotations.RoutingAddressesExternal("")),
-		annotations.Remove(annotations.RoutingAddressesPublic("")),
-	}
+func (r *Controller) propagateHostsToWatchedCR(ctx context.Context, target *unstructured.Unstructured, publicHosts, externalHosts []string) error {
+	err := unstruct.PatchWithRetry(ctx, r.Client, target, func() error {
+		// Always remove the annotations first
+		annotations.Remove(annotations.RoutingAddressesExternal(""))(target)
+		annotations.Remove(annotations.RoutingAddressesPublic(""))(target)
```

Comment:
Why not having them as initial list of meta options which then is appended like we did before?

Reply:
true, this change isn't necessary, some holdover from initial bug fixing. reverted this

Reply:
a65f9b071869b9f020886831de888b35eb1f7736